src/zenml/integrations/kubeflow/utils.py

```diff
@@ -61,3 +61,7 @@ def apply_pod_settings(
     resource_limits = settings.resources.get("limits") or {}
     for name, value in resource_limits.items():
         container_op.add_resource_limit(name, value)
+
+    annotations = settings.annotations or {}
+    for name, value in annotations.items():
```

Comment:
```suggestion
    for name, value in settings.annotations.items():
```
Seems to be a non-optional Dict and this should work as well right?